lib/URDNA2015.js

```diff
@@ -99,6 +100,24 @@ module.exports = class URDNA2015 {
       // 5.4.5) Set simple to true.
     }
 
+    if(this.maxDeepIterations < 0) {
+      // calculate maxDeepIterations if not explicit
+      if(this.maxWorkFactor === 0) {
+        // use 0 default
```

Comment:
I found this comment a bit confusing ... `maxDeepIterations` is less than `0` right now ... so should it be set to `0`? What does "use 0 default" mean?

It looks like `this.remainingDeepIterations` will be set to `-1` in that case ... maybe the code on lines 155-119 that sets it should be moved inside of the `this.maxDeepIterations < 0` conditional ... either that or set `maxDeepIterations` to `0` explicitly?

Reply:
A refactoring broke the logic here.  Fixed it and added some tests.